user/tests/wiring/no_fixture/cellular.cpp

```diff
@@ -157,6 +157,56 @@ test(CELLULAR_07_rssi_is_valid) {
     assertMoreOrEqual(s.rssi, -150);
 }
 
+test(CELLULAR_08_sigstr_is_valid) {     // collects signal strength and quality
```

Comment:
Tested this on Boron LTE, and seems reflective of the changes. I used certain values and debug statements to verify we are seeing some values. Also, interestingly, the RAT is `6 (NET_ACCESS_TECHNOLOGY_LTE)`

```
Test CELLULAR_07_rssi_is_valid passed.
StrV: -95.00
QualV: -15.50
RAT: 6
Test CELLULAR_08_sigstr_is_valid passed.
Test CLOUD_01_Particle_Connect_Does_Not_Block_In_SemiAutomatic_Mode passed.
```

Reply:
Probably a separate topic - but I think the RAT just reads 6 for M1? I printed `sigext rat` below in the log.
```
0000060224 [ncp.at] TRACE: > AT+COPS?
0000060272 [ncp.at] TRACE: < +COPS: 0,2,"310410",7
0000060275 [ncp.at] TRACE: < OK
0000060276 [ncp.at] TRACE: > AT+UCGED=5
0000060322 [ncp.at] TRACE: < OK
0000060323 [ncp.at] TRACE: > AT+UCGED?
0000060372 [ncp.at] TRACE: < +RSRP: 314,5110,"-094.00",
0000060375 [ncp.at] TRACE: < +RSRQ: 314,5110,"-13.20",
0000060377 [ncp.at] TRACE: < OK
0000060378 [wiring] INFO: sigext rat: 6
```

Reply:
This is weird, because the AT command reference states:
```
SARA-R410M-02B•   <AcT>=8 is used for LTE Cat M1 on SARA-R410M-02B-00.
```

Here we can see AcT=7 for some reason.

Reply:
Ah, but it also says:
```
SARA-R410M-02B / SARA-R410M-52B - 7, 9
```

Reply:
For SARA R4 we should set `NET_ACCESS_TECHNOLOGY_LTE_CAT_M1` if AcT is 7 or 8, I think that's the solution and `NET_ACCESS_TECHNOLOGY_LTE_CAT_NB1` if it's 9.

Reply:
Sounds about right.

Reply:
This is probably where we need to make this change - https://github.com/particle-iot/device-os/blob/f6a0a5970c90284c07e078b9894ca0b8566f0fad/hal/network/ncp/cellular/cellular_hal.cpp#L85

Reply:
I would actually patch it here depending on the modem type: https://github.com/particle-iot/device-os/blob/develop/hal/src/boron/network/sara_ncp_client.cpp#L471 

And for Gen 2:
https://github.com/particle-iot/device-os/blob/develop/hal/src/electron/modem/mdm_hal.cpp#L1718
https://github.com/particle-iot/device-os/blob/develop/hal/src/electron/modem/mdm_hal.cpp#L496

Again depending on the modem type.

Reply:
This can be done as a separate PR.

Reply:
Coolio!